clients/java/signalr/src/main/java/JsonHubProtocol.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+
+public class JsonHubProtocol implements IHubProtocol{
+    private JsonParser jsonParser = new JsonParser();
```

Comment:
this could be `final` (as well as almost any local variable below and function input parameters)